moduledeps/module.go

```diff
@@ -110,11 +110,11 @@ func (s sortModules) Swap(i, j int) {
 func (m *Module) ProviderRequirements() discovery.PluginRequirements {
 	ret := make(discovery.PluginRequirements)
 	for pFqn, dep := range m.Providers {
-		providerStr := pFqn.LegacyString()
-		if existing, exists := ret[providerStr]; exists {
-			ret[providerStr].Versions = existing.Versions.Append(dep.Constraints)
+		providerType := pFqn.Type
```

Comment:
This is a "load-bearing" change that makes no _practical_ difference beyond removing references to legacy strings.